src/functions/avl-validate/index.ts

```diff
@@ -0,0 +1,157 @@
+                        ...verifiedServiceRequest.ServiceRequest.VehicleMonitoringRequest.VehicleMonitoringRequest,
+                    },
+                },
+            },
+        },
+    };
+
+    const builder = new XMLBuilder({
+        ignoreAttributes: false,
+        format: true,
+        attributeNamePrefix: "@_",
+    });
+
+    const request = builder.build(completeObject) as string;
+
+    return request;
+};
+
+const parseXml = (xml: string) => {
+    const parser = new XMLParser({
+        allowBooleanAttributes: true,
+        ignoreAttributes: false,
+        parseTagValue: true,
+    });
+
+    const parsedXml = parser.parse(xml) as Record<string, unknown>;
+
+    const parsedJson = avlServiceDeliverySchema.safeParse(parsedXml.Siri);
+
+    if (!parsedJson.success) {
+        logger.error(
+            "There was an error parsing the service delivery response from the data producer.",
+            parsedJson.error.format(),
+        );
+        throw new InvalidXmlError();
+    }
+
+    return parsedJson.data;
+};
+
+export const handler = async (event: APIGatewayEvent): Promise<APIGatewayProxyResult> => {
+    try {
+        logger.info("Starting AVL feed validator");
+
+        const { url, username, password } = requestBodySchema.parse(event.body);
+
+        const requestTime = getDate();
+        const currentTime = requestTime.toISOString();
+
+        const serviceRequest = generateServiceRequestMessage(currentTime);
+
+        const serviceDeliveryResponse = await axios.post<string>(url, serviceRequest, {
+            headers: {
+                "Content-Type": "text/xml",
+                Authorization: createAuthorizationHeader(username, password),
+            },
+        });
+
+        const serviceDeliveryBody = serviceDeliveryResponse.data;
+
+        if (!serviceDeliveryBody) {
+            logger.warn("No body was returned from data producer");
+            return createValidationErrorResponse(["No body was returned from the data producer"]);
+        }
+
+        const parsedServiceDeliveryBody = parseXml(serviceDeliveryBody);
+
+        const status = parsedServiceDeliveryBody?.ServiceDelivery.Status ?? null;
+
+        const siriVersion = parsedServiceDeliveryBody?.["@_version"] ?? null;
+
+        if (!status || status !== "true") {
+            logger.warn("Data producer did not return a status of true");
+            return createValidationErrorResponse(["Data producer did not return a status of true"]);
+        }
+
+        return { statusCode: 200, body: JSON.stringify({ siriVersion }) };
+    } catch (error) {
+        if (error instanceof ZodError) {
+            logger.warn("Invalid request", error);
+            return createValidationErrorResponse(error.errors.map((error) => error.message));
+        }
+
+        if (error instanceof AxiosError) {
+            logger.warn("Invalid request", error);
+            return createValidationErrorResponse([`Invalid request: ${error.message}`]);
```

Comment:
I'm not sure passing through the error message is the best option here, perhaps we provide a more generic message?